python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -1,36 +1,101 @@
 import datetime
 from abc import ABC, abstractmethod, abstractproperty
 from collections import defaultdict
+from enum import Enum
 from typing import (
     TYPE_CHECKING,
     AbstractSet,
+    Dict,
+    FrozenSet,
     Mapping,
     NamedTuple,
     Optional,
+    Sequence,
+    Set,
+    Tuple,
+    cast,
 )
 
+import dagster._check as check
 from dagster._core.definitions.data_time import CachingDataTimeResolver
 from dagster._core.definitions.events import AssetKey, AssetKeyPartitionKey
 from dagster._core.definitions.freshness_based_auto_materialize import (
-    freshness_conditions_for_asset_key,
+    freshness_evaluation_results_for_asset_key,
 )
 from dagster._core.definitions.partition_mapping import IdentityPartitionMapping
 from dagster._core.definitions.time_window_partition_mapping import TimeWindowPartitionMapping
-from dagster._serdes.serdes import whitelist_for_serdes
+from dagster._serdes.serdes import (
+    NamedTupleSerializer,
+    UnpackContext,
+    UnpackedValue,
+    WhitelistMap,
+    whitelist_for_serdes,
+)
 from dagster._utils.caching_instance_queryer import CachingInstanceQueryer
 
-from .asset_graph import AssetGraph
-from .auto_materialize_condition import (
-    AutoMaterializeCondition,
-    AutoMaterializeDecisionType,
-    MissingAutoMaterializeCondition,
-    ParentMaterializedAutoMaterializeCondition,
-    ParentOutdatedAutoMaterializeCondition,
-)
+from .asset_graph import AssetGraph, sort_key_for_asset_partition
+from .partition import SerializedPartitionsSubset
 
 if TYPE_CHECKING:
     from dagster._core.definitions.asset_daemon_context import AssetDaemonContext
     from dagster._core.definitions.asset_daemon_cursor import AssetDaemonCursor
+    from dagster._core.instance import DynamicPartitionsStore
+
+
+@whitelist_for_serdes
+class AutoMaterializeDecisionType(Enum):
+    """Represents the set of results of the auto-materialize logic.
+
+    MATERIALIZE: The asset should be materialized by a run kicked off on this tick
+    SKIP: The asset should not be materialized by a run kicked off on this tick, because future
+        ticks are expected to materialize it.
+    DISCARD: The asset should not be materialized by a run kicked off on this tick, but future
+        ticks are not expected to materialize it.
+    """
+
+    MATERIALIZE = "MATERIALIZE"
+    SKIP = "SKIP"
+    DISCARD = "DISCARD"
+
+
+class AutoMaterializeRuleEvaluationData(ABC):
+    ...
+
+
+@whitelist_for_serdes
+class GenericRuleEvaluationData(
+    AutoMaterializeRuleEvaluationData,
+    NamedTuple("_GenericRuleEvaluationData", [("description", str)]),
+):
+    ...
+
+
+@whitelist_for_serdes
+class ParentUpdatedRuleEvaluationData(
+    AutoMaterializeRuleEvaluationData,
+    NamedTuple(
+        "_ParentUpdatedRuleEvaluation",
+        [("updated_keys", FrozenSet[AssetKey]), ("will_update_keys", FrozenSet[AssetKey])],
+    ),
+):
+    ...
```

Comment:
Prefer `pass` to `...` in this case, because the latter implies that something is omitted.